323 proposal reduce ranging through batch entries

bkmoovio opened the pull request:
I'm going to submit a pull request for 323.  It covers all but IAT, I kind of had to do that to remove the isAddendaCount function.  I'll make a separate issue for IAT. 

I did make some generic functions for returning error message if a batch had an incorrect TypeCode for an Addenda.  I wanted that to be caught at the batch level.  

Take a look and let me know what you think.  I can tear it down if you don't like it and we can do it another way, but having a function with switches for SEC Code or typeCode assertion doesn't seem better than handling the conditions within the batchSEC type itself.   Either way we may want to look into batch and the ranging of entries for other functions as well.   Thanks in advance.

bkmoovio committed:
323

BatchWeb and BatchShr
Make generic message in batcher for msgBatchRequiredAddendaCount

bkmoovio committed:
323

batchPOS

bkmoovio committed:
323

batchCCD
Updates to batchPOS, batchSHR and batchWeb

bkmoovio committed:
323

batchPPD

Updates for batchWeb and batchWEB_test

96.8% code coverage

bkmoovio committed:
323

Add Category Logic for batchPPD, batchWEB, and batchTEL
Removed dead code - isPaymentTypeCode

bkmoovio committed:
323

Remove var

bkmoovio committed:
323

batchPOS and batchSHR

bkmoovio committed:
323

batchCCD and batchCCD test
batchSHR and batchPOS require one Addenda02

bkmoovio committed:
323

batchARC, batchBOC, and batchCIE

bkmoovio committed:
323

323 batchCIE, batchPOP, batchRCK

bkmoovio committed:
Merge branch 'master' into 323-Proposal-Reduce-ranging-through-batch-entries

bkmoovio committed:
323

Update batch to have validation checks for:
CategoryForwardAddenda05
CategoryForwardAddenda98
CategoryForwardAddenda99

bkmoovio committed:
323

PPD, RCK., SHR, TEL updates to use new batch fucntions

bkmoovio committed:
323

Updtes for batch function calls

bkmoovio committed:
323

batchCTX

bkmoovio committed:
323

Adjust counter for AddendaCount

bkmoovio committed:
323 CTX test changes

323 CTX test changes

adamdecaf commented on batch.go:
```diff
@@ -422,16 +407,34 @@ func (batch *batch) isCategory() error {
-		}
+// categoryForwardAddenda02 verifies CategoryForward Addenda02 TypeCode is 02
+func (batch *batch) categoryForwardAddenda02(entry *EntryDetail, addenda Addendumer) error {
+	return nil
```
This needs an implementation right?

wadearnold replied:
This seems like test cases should fail. Otherwise, we need some test cases that do fail!

bkmoovio committed:
323 categoryForwardAddenda02

Add implementation of categoryForwardAddenda02
Tests were already written for the implementation but never hit the failure condition.  e.g. testBatchPOSInvalidAddendum
